STL_Extension/include/CGAL/STL_Extension/internal/mesh_option_classes.h

```diff
@@ -165,6 +168,116 @@ struct Features_options
   bool b_;
 };
 
+template <typename Value>
+class Input_const_iterator_interface
+{
+public:
+  virtual ~Input_const_iterator_interface() {}
+  virtual const Value& operator*() = 0;
+  virtual Input_const_iterator_interface<Value>* operator++() = 0;
+  virtual bool operator!=(const Input_const_iterator_interface<Value>* other) const = 0;
+  virtual Input_const_iterator_interface<Value>* clone() = 0;
+};
+
+template <typename Value, typename Iterator>
+struct Input_const_iterator_container
+  : Input_const_iterator_interface<Value>
+{
+  typedef Input_const_iterator_container<Value, Iterator> Self;
+public:
+  Input_const_iterator_container(const Iterator& it) : it_(it) {}
+
+  virtual ~Input_const_iterator_container() {}
+
+  virtual const Value& operator*()
+  {
+    return *it_;
+  }
+
+  virtual Input_const_iterator_interface<Value>* operator++()
+  {
+    ++it_;
+    return this;
+  }
+
+  virtual bool operator!=(const Input_const_iterator_interface<Value>* other) const
+  {
+    const Self* other_casted = dynamic_cast<const Self*>(other);
+    if (other_casted == nullptr)
+      return true;
+    return it_ != other_casted->it_;
+  }
+
+  virtual Input_const_iterator_interface<Value>* clone()
+  {
+    return new Input_const_iterator_container<Value, Iterator>(it_);
+  }
+
```

Comment:
There is almost certainly a memory leak, here, because I do not see any corresponding `delete`.

Reply:
@ange-clement did you fix that somewhere?